### DISCUSSION TITLE
Getting Type of WebSocket Object (Best Practices)

### DISCUSSION THREAD

**Discussion by hellmannx** [uNetworking/uWebSockets#1346] [Q&A]
I need to store websockets in an array for each connection made. 

    // Needed to go through the source files.
    static std::vector<uWS::WebSocket<false, true, DataType> *> users; 

    app.ws<DataType>("/ws", {
        
        .open = [](auto * ws){
            users.push_back(ws);
             // This is also possible, but not accessible inside .message
             static std::vector<decltype(ws)> users2;
            ws->subscribe("sensors/+/house");
        },

        .close = [](auto *ws, int code, std::string_view message){
            users.erase(std::remove(users.begin(), users.end(), ws), users.end());
            std::cout << "Client disconnected!" << std::endl;
        },

        .message = [](auto *ws, std::string_view message, uWS::OpCode opCode){
            try{
                std::string message2 = std::string(message) + std::string(" ACK");
                for(const auto & ws2 : users)
                    if(ws != ws2)
                        ws2->send(message2, opCode);

            }catch(std::exception& e){
                std::cout << e.what() << std::endl;
            }
        },
    });

My problem is I had to go through source files and find `uWS::WebSocket<false, true, DataType> *` declaration to declare my array for websocket connection while lambdas are declared with the `auto` keyword in the examples.

Is there a way to address type of websocket object without having to go through the source files?

**Comment by hakoptak** [comment]
Hi, I can totally relate to that. This is exactly what I did. I think this is more of a documentation thing.

Suggestion: maybe add an example for this use case?
